test/emu_mps/test_probabilitysample.py

```diff
@@ -1,30 +1,49 @@
+import pytest
 import torch
 
 from emu_mps import MPS
 
-from test.utils_testing import ghz_state_factors
+from test.utils_testing import cpu_multinomial_wrapper, ghz_state_factors
+
+from unittest.mock import patch
 
 
 seed = 1337  # any number will do
-device = "cpu"
+device = "cpu"  # 'cuda'
+
 dtype = torch.complex128
 
 
-def test_sampling_ghz5_mps():
-    device = "cpu"
+@pytest.mark.parametrize(
+    "basis",
+    (
+        ("0", "1"),
```

Comment:
just (r, g) as well to check the full space ? though the sampling part is expected do to the same